src/pymorize/units.py

```diff
@@ -110,15 +98,24 @@ def handle_unit_conversion(da: xr.DataArray, rule: Rule) -> xr.DataArray:
     model_unit = rule.get("model_unit")
```

Comment:
About the function `handle_unit_conversion(da: xr.DataArray, rule: Rule)` would it make sense to have a 3rd argument `dryrun=False` so that this function can be used before computation in a loop over the rules, to report wrong units in the model side, missing mappings for the tables, etc?

Reply:
We could do this as long as it has a default argument of `dryrun=False`, otherwise we would break the pipeline API.

Reply:
Correct, I would set the parameter to be always `False`, unless specified otherwise. I suggested `dryrun` as name but if you rather have something like `check` that's also a good name for me.

Reply:
I think instead of the option `dryrun=False` which breaks the API, the same thing can be achieved through  a command line option with a meaningful name like "check" or "validate". At the moment, the check for frequency of source data and the frequency that the data needs to be converted to (according to frequency mention in table) is implemented in `CMORizer._post_init_checks`. The unit conversion check can also be added there. The command line option can skip dask cluster creating thing, construct the CMORizer object by populating the rules so that `_post_init_checks` has all the information it needs to verify and validate the user inputs for errors.

Reply:
> Will the following break the pipeline API? Why?
At the moment, any step in a pipeline **must** have the signature:
```python
def my_step(data, rule):
   ...
   return data
```

I think it would be cleanest to turn `_post_init_checks` into a main CMORizer object method (e.g. `validate`), which you can run independently of `process`. In my view, constructing the CMORizer object should still be possible, even if you fill it will rubbish rules (this makes testing easier). So, something like this:

```python
cmorizer = CMORizer.from_dict(cfg)
cmorizer.validate()
cmorizer.process()
```

Reply:
After discussion with @pgierz the dryrun capabilities will be most likely implemented via a decorator, or simply by a validation schema of the steps that uses try, catches exceptions and gathers errors before computation. Consequently, this suggestion does not belong to this PR and can be omitted.